enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,852 @@
+default.
+
+```yaml
+metadata:
+  annotations:
+    workload.openshift.io/management: |
+      {"effect": "PreferredDuringScheduling"}
+```
+
+The admission hook will reject pods with multiple annotations, for
+now. Future work may allow multiple workload types with different
+priorities to support clusters with different types of configurations.
+
+#### Pod mutation
+
+The kubelet and API admission hook will change pods annotated with
+`workload.openshift.io/management` so the CPU requests are replaced
+with management CPU requests and an annotation is added with the same
+value.
+
+```yaml
+requests:
+  cpu:
+    400m
+```
+
+becomes
+
+```yaml
+requests:
+  management.workload.openshift.io/cores: 400
+```
+
+and
+
+```yaml
+annotations:
+  io.openshift.workload.management.cpushares/{container-name}: 400
+```
+
+The annotation name used to set the resource requests is reversed
+(when compared to the other workload annotation or the resource
+request name) so that CRI-O can be configured with a prefix value to
+simplify the way it processes the strings without having to understand
+their content.
+
+The new request value and annotation value are scaled up by 1000 from
+the original CPU request input because opaque resources do not support
+units or fractional values.
+
+As a special case, kubelet and the API admission hook should strip any
+annotations from pods trying to set the "cpuset" resource. See the
+next section for more details.
+
+#### CRI-O Changes
+
+CRI-O will be updated to support new configuration settings for
+workload types.
+
+```ini
+[crio.runtime.workloads.{workload-type}]
+  activation_annotation = "workload.openshift.io/{workload-type}"
+  resource_annotation_prefix = "io.openshift.workload.{workload-type}"
+  resources = {
+    "cpu": "",
+    "cpuset": "0-1",
+  }
+```
+
+The `activation_annotation` field is used to match pods that should be
+treated as having the workload type. The annotation key on the pod is
+compared for an exact match against the value specified in the
+configuration file.
+
+The `resource_annotation_prefix` is the start of the annotation key
+used to pass settings from the admission hook down to CRI-O.
+
+The `resources` map associates annotation suffixes with default
+values. CRI-O will define a well-known set of resources and other
+values will not be allowed. In OpenShift, we do not want to allow
+arbitrary pods to set their own `cpuset` but other CRI-O users may
+want that ability.
+
+To pass a setting into CRI-O, the pod should have an annotation made
+by combining the `resource_annotation_prefix`, the key from
+`resources`, and the container name, like this:
+
+```text
+io.openshift.workload.management.cpushares/container_name = {value}
+```
+
+In the management workload case, we will configure it with values like
+
+```ini
+[crio.runtime.workloads.management]
+  activation_annotation = "workload.openshift.io/management"
+  resource_annotation_prefix = "io.openshift.workload.management"
```

Comment:
did this change from `annotation_prefix` ?

Reply:
Fixed